proxy/request_test.go

```diff
@@ -0,0 +1,25 @@
+package proxy
+
+import (
+	"testing"
+)
+
+func TestBadMessage(t *testing.T) {
+	req := ""
+	resp := handleRequest([]byte(req))
+
+	expected := `{"id":null,"error":{"errcode":"M_NOT_JSON","error":"unexpected end of JSON input"}}`
+	if string(resp) != expected {
```

Comment:
The reason I didn't want to do that was because I wanted to check that "id" was null, and "result" was absent, which I can't really do if I add another layer of json (un)marshalling. But yes, I agree.

Maybe looking for particular substrings in the response would be less awful?